src/dock-menu.ts

```diff
@@ -1,22 +1,25 @@
-import { Menu, BrowserWindow, MenuItemConstructorOptions } from 'electron'
+import {
+  BrowserWindow,
+  Menu,
+  MenuItemConstructorOptions,
+  ipcRenderer as ipc
+} from 'electron'
 
 const baseGmailUrl = 'https://mail.google.com'
 
 export function buildDockMenu(mainWindow: BrowserWindow): Menu {
-  const inboxButton = createInboxButton(mainWindow)
+  const inboxButton = createInboxButton()
   const composeButton = createComposeButton(mainWindow)
   const sentButton = createSentButton(mainWindow)
   return Menu.buildFromTemplate([inboxButton, composeButton, sentButton])
 }
 
-function createInboxButton(
-  mainWindow: BrowserWindow
-): MenuItemConstructorOptions {
+function createInboxButton(): MenuItemConstructorOptions {
   return {
     label: 'Inbox',
-    async click() {
-      await mainWindow.loadURL(`${baseGmailUrl}/#inbox`)
-      mainWindow.show()
+    click() {
+      // TODO: investigate why ipc is undefined here
+      ipc.send('open-inbox-view')
```

Comment:
@timche 

I've started on the `ipc` method of doing this as you suggested, but I'm running into an issue where `ipc` here is always undefined. It looks like the import isn't working properly. I've ran a few searches but they all seemed to point me toward
 
* removing `node_modules` and reinstalling (tried that)
* not correctly importing it (followed the documentation)

I've also noticed that you use it in `preload.ts`, but I haven't been able to confirm whether it works there or not, since I don't think it's actually being run/used AFAIK.

Any thoughts?

Reply:
1. You are importing `ipcRenderer` which is only supposed to be imported in the renderer and not the main process.
2. To send a message from the main process to the renderer, you need use [BrowserWindow.webContents.send()](https://www.electronjs.org/docs/api/web-contents#contentssendchannel-arg1-arg2-).
3. We have a helper function in [`utils.ts`](https://github.com/timche/gmail-desktop/blob/master/src/utils.ts) called [`sendChannelToMainWindow()`](https://github.com/timche/gmail-desktop/blob/master/src/utils.ts#L23:L28) that you can use instead.